src/collectors/include_graph/include_graph.cpp

```diff
@@ -56,7 +57,7 @@ class IncludeGraphImpl {
   clang::ast_matchers::StatementMatcher sm1 =
       clang::ast_matchers::declRefExpr().bind("ref");
   clang::ast_matchers::TypeLocMatcher sm2 =
-      clang::ast_matchers::typeLoc().bind("type");
+      clang::ast_matchers::typeLoc(optionally(loc((const internal::Matcher<clang::QualType> &)hasDeclaration(decl().bind("decl"))))).bind("type");
```

Comment:
To confirm, the `hasDeclaration()` matcher can find the using statement, but `type->getDecl()` doesn't?

Reply:
The current code doesn't call type->getDecl() but tries to guess the Decl

https://github.com/bloomberg/clangmetatool/blob/cb17e415ffbc6b98c2f307f2aaff4be5cd56af0a/src/collectors/include_graph/include_graph_util.cpp#L240-L255

Reply:
I just think its confusing that we try to get the decl from the matcher and during the postprocessing, is it possible to get rid of the extract_decl_for_type calls?

Reply:
Agree. I am/was worried that there may be cases that the matcher does not catch.